diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/VariableSearchASTUtils.kt

```diff
@@ -0,0 +1,101 @@
+package org.cqfn.diktat.ruleset.utils
+
+import com.pinterest.ktlint.core.ast.ElementType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getChildrenOfType
+import org.jetbrains.kotlin.psi.psiUtil.getParentOfType
+import org.jetbrains.kotlin.psi.psiUtil.parents
+import org.jetbrains.kotlin.psi.psiUtil.referenceExpression
+
+/**
+ * [this] - root node of a type File that is used to search all declared properties (variables)
+ * and it's usages (rvalues).
+ * (!) ONLY for nodes of File elementType
+ *
+ * @return a map of a property to it's usages
+ */
+fun ASTNode.collectAllDeclaredVariablesWithUsages(): Map<KtProperty, List<KtNameReferenceExpression>> {
+    val fileNode = this.psi
+    require(fileNode is KtFile) {
```

Comment:
Why not just make this method an extension on `KtFile` receiver?

Reply:
Because we are operating with ASTNode all over the code. I have changed require statement